lib/dialects/abstract/query-generator.js

```diff
@@ -1192,9 +1192,10 @@ var QueryGenerator = {
                          primaryKeysLeft[0]
 
             , tableRight = as
+            , keyRight  = association.options ? association.options.otherKey || primaryKeysRight[0] : primaryKeysRight[0]
```

Comment:
I'm afraid that's not what I'm seeing.  `association.options.otherKey` is present when provided in the belongsTo relationship, but there is no `association.otherKey` attribute getting set.